services/slam/builtin/builtin_test.go

```diff
@@ -1105,5 +1109,60 @@ func createTempFolderArchitecture() (string, error) {
 
 func resetFolder(path string) error {
 	err := os.RemoveAll(path)
+	if err != nil {
+		return err
+	}
+	err = os.Mkdir(path, os.ModePerm)
 	return err
 }
+
+func checkDeleteProcessedData(t *testing.T, mode slam.Mode, dir string, prev int, deleteProcessedData, online bool) int {
+	var numFiles int
+
+	switch mode {
+	case slam.Mono:
+		numFilesRGB, err := checkDataDirForExpectedFiles(t, dir+"/data/rgb", prev, online, deleteProcessedData)
+		test.That(t, err, test.ShouldBeNil)
+
+		numFiles = numFilesRGB
+	case slam.Rgbd:
+		numFilesRGB, err := checkDataDirForExpectedFiles(t, dir+"/data/rgb", prev, online, deleteProcessedData)
+		test.That(t, err, test.ShouldBeNil)
+
+		numFilesDepth, err := checkDataDirForExpectedFiles(t, dir+"/data/depth", prev, online, deleteProcessedData)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, numFilesRGB, test.ShouldEqual, numFilesDepth)
+		numFiles = numFilesRGB
+	case slam.Dim2d:
+		numFiles2D, err := checkDataDirForExpectedFiles(t, dir+"/data", prev, online, deleteProcessedData)
+		test.That(t, err, test.ShouldBeNil)
+		numFiles = numFiles2D
+	default:
+	}
+	return numFiles
+}
+
+func checkDataDirForExpectedFiles(t *testing.T, dir string, prev int, delete_processed_data, online bool) (int, error) {
```

Comment:
Since. I am adding an error statement based on your other comment above for the `delete_processed_data == true` and `offline == true`, can I assume this comment is covered?

Reply:
yep!